### PR TITLE
SF-2527 Support for invalid reattached notes

### PR DESCRIPTION
**Pull request by pmachapman**
This PR allows proper error handling for invalid reattached notes.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2334)

### CODE REVIEW

File: src/SIL.XForge.Scripture/Services/ParatextService.cs [hunk 1]
```diff
@@ -3180,11 +3178,18 @@
...
+                verseRef = new VerseRef(reattachedParts[0]);
+                selectedText = reattachedParts[1];
+                startPos = int.Parse(reattachedParts[2]);
+                contextBefore = reattachedParts[3];
+                contextAfter = reattachedParts[4];
+            }
+            catch (Exception)
+            {
+                // Ignore any errors parsing the re-attached verse
+            }
```
[the remark this answers is not in the document]
Comment (Nateowami) [reply src/SIL.XForge.Scripture/Services/ParatextService.cs:3192]: Is there a good way of using Paratext's own internal logic to handle this? Or is that more effort than it's worth? In general I think we don't want to re-implement things, but obviously sometimes it's the simpler option.

[the remark this answers is not in the document]
Comment (pmachapman) [reply src/SIL.XForge.Scripture/Services/ParatextService.cs:3192]: I have updated it to use the ParatextData logic - I discovered my problems using it were due to the test not matching what Paratext did when reattaching a note with invalid reattach location.

**Review by pmachapman — COMMENTED** [review]
I have shared the ALBT project with the testers, and given you access, too.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2334#-:-NrWjM6M60C7BCKyy-li:bj1grui)* status: all files reviewed, 2 unresolved discussions (waiting on @github-advanced-security[bot] and @RaymondLuong3)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.ts` line 476 at r2](https://reviewable.io/reviews/sillsdev/web-xforge/2334#-NrMqVdnC4CsJpG3PjNi:-NrWl1GQCxo3LiTRwwGi:b-896fix) ([raw file](https://github.com/sillsdev/web-xforge/blob/e91a1458baa2bcda519248fa4b7be101b94e09f6/src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.ts#L476)):*
<details><summary><i>Previously, RaymondLuong3 (Raymond Luong) wrote…</i></summary><blockquote>

Can you add a comment here explaining why we return early?
</blockquote></details>

Done.

**Review by RaymondLuong3 — APPROVED** [review]
Thanks Peter. This looks ready to go to testing.

> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2334#-:-Nraf6ct-KJNefGXoxiF:b-9bwii7)* status: all files reviewed, 1 unresolved discussion (waiting on @github-advanced-security[bot])

**Pull request merged**